gitmap.go

```diff
@@ -104,8 +110,14 @@ func Map(repository, revision string) (*GitRepo, error) {
 	return &GitRepo{Files: m, TopLevelAbsPath: topLevelPath}, nil
 }
 
-func git(args ...string) ([]byte, error) {
-	out, err := exec.Command(gitExec, args...).CombinedOutput()
+func git(opts Options, args ...string) ([]byte, error) {
+	var cmd *exec.Cmd
+	if opts.GetGitCommandFunc != nil {
```

Comment:
I would prefer if we did this in the `Map` func and set the value on the `Options` struct; that way we don't have to think about it if we add other git operations (the opt.GetGitCommandFunc is always set).

Reply:
Sorry, I did not read your previous comment carefully enough.